graphql/resolvers/addComment.ts

```diff
@@ -10,13 +19,70 @@ export const addComment = async (
   const { line, submissionId, fileName, content } = arg
   const authorId = ctx.req.user?.id
   if (!authorId) throw new Error('No authorId field')
-  return prisma.comment.create({
+
+  const comment = await prisma.comment.create({
     data: {
       line,
       submissionId,
       authorId,
       fileName,
       content
+    },
+    include: {
+      author: {
+        select: {
+          id: true,
+          username: true,
+          discordId: true,
+          name: true
+        }
+      },
+      submission: {
+        include: {
+          user: {
+            select: {
+              id: true,
+              username: true,
+              discordId: true
+            }
+          },
+          lesson: true,
+          challenge: true
+        }
+      }
     }
   })
+
+  const { author, submission } = comment
+
+  if (author.id !== submission.user.id && submission.user.discordId) {
+    const notificationEmbed: MessageEmbedOptions = {
```

Comment:
Is it necessary to explicitly give it a type? TypeScript is capable of inferring its type and comparing the object's properties with `sendDirectMessage` first parameter type.

Reply:
Yeah ts will figure out if the type is sufficient when this object is passed to the function, but I have explicitly typed the object literal as `MessageEmbedOptions` to help with autocomplete and ides when reading/modifying the variable where it is declared.